src/common/modules/UnicodeFontHandler.js

```diff
@@ -154,75 +143,60 @@ function handle(info, tab) {
  * @returns {void}
  */
 function applySettings(unicodeFont) {
-	const menus = browser.menus || browser.contextMenus; // fallback for Thunderbird
-
-	menus.removeAll();
-
-	if (unicodeFont.changeCase) {
-		for (const id of caseIds) {
-			// .replaceAll(" ", "-");
-			const aid = id.toLowerCase().split(" ").join("-");
-			menus.create({
-				"id": aid,
-				"title": changeCase[aid](id),
-				"contexts": ["editable"],
-			});
-		}
-	}
-
-	if (unicodeFont.changeCase && unicodeFont.changeFont) {
-		menus.create({
-			id: "separator-2",
-			type: "separator",
-			contexts: ["editable"]
-		});
-	}
-
-	if (unicodeFont.changeFont) {
-		for (const id of fontIds) {
-			if (id === "separator") {
-				menus.create({
-					// id: id,
-					type: "separator",
-					contexts: ["editable"]
-				});
-			} else {
-				// .replaceAll(" ", "-");
-				const aid = id.toLowerCase().split(" ").join("-");
-				// console.log(id, aid, fonts[aid], changeFont(id, aid));
-				menus.create({
-					"id": aid,
-					"title": changeFont(id, aid),
-					"contexts": ["editable"],
-				});
-			}
-		}
-	}
+    const menus = browser.menus || browser.contextMenus; // fallback for Thunderbird
+
+    menus.removeAll();
+
+    let menuItemsToShow = contextMenuList;
+    if (!unicodeFont.changeCase) {
+        const caseValues = Object.values(CASE);
+        menuItemsToShow = contextMenuList.filter((id) => !caseValues.includes(id) );
+    }
+    if (!unicodeFont.changeFont) {
+        const fontValues = Object.values(FONT);
+        menuItemsToShow = contextMenuList.filter((id) => !fontValues.includes(id) );
+    }
+    menuItemsToShow
+        .filter((id, index) => (index !== 0 || id !== SEPARATOR))
```

Comment:
This does not work if the user only enables the change case options, as they get a bunch of separators at the bottom: 
![image](https://user-images.githubusercontent.com/15305150/100961512-67dbc080-34d7-11eb-83f7-0a3b656bc2be.png)